v2/cli/init_command.go

```diff
@@ -0,0 +1,278 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/brigadecore/brigade/sdk/v2/core"
+	"github.com/brigadecore/brigade/v2/internal/file"
+	"github.com/urfave/cli/v2"
+)
+
+var initCommand = &cli.Command{
+	Name:  "init",
+	Usage: "Initialize a new Brigade project",
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:    flagID,
+			Aliases: []string{"i"},
+			Usage: "Name of the Brigade project to initialize " +
+				"(required)",
+			Required: true,
+		},
+		&cli.StringFlag{
+			Name:    flagLanguage,
+			Aliases: []string{"l"},
+			Usage: "Specify the scripting language to use for handling events" +
+				" -- JavaScript (js) or TypeScript (ts)",
+			Value: "ts",
+		},
+		&cli.StringFlag{
+			Name:    flagGit,
+			Aliases: []string{"g"},
+			Usage: "Optionally specify a git repository where the" +
+				" event handling script can be found",
+		},
+	},
+	Action: initialize,
+}
+
+// nolint: gocyclo
+func initialize(c *cli.Context) error {
+
+	// Initialize fields to fill in the template yaml files
+	fields := struct {
+		ProjectID   string
+		Language    string
+		GitCloneURL string
+		Script      string
+	}{
+		ProjectID:   c.String(flagID),
+		Language:    strings.ToLower(c.String(flagLanguage)),
+		GitCloneURL: c.String(flagGit),
+		Script:      "",
+	}
+
+	err := core.ValidateProjectID(fields.ProjectID)
+	if err != nil {
+		return err
+	}
+
+	// Check if input language is valid
+	fields.Language, err = fileExtensionForLanguage(fields.Language)
+	if err != nil {
+		return err
+	}
+
+	// Check if the .brigade directory exists
+	err = file.EnsureDirectory(".brigade")
+	if err != nil {
+		return err
+	}
+
+	// Generate script based on given language
+	var scriptBytes []byte
+	switch fields.Language {
+	case "ts":
+		scriptBytes, err = execTemplate(typeScriptTemplate, fields)
+	case "js":
+		scriptBytes, err = execTemplate(javaScriptTemplate, fields)
+	}
+	if err != nil {
+		return err
+	}
+	fields.Script = string(scriptBytes)
+
+	brigadeScriptFileName := fmt.Sprintf("brigade.%s", fields.Language)
+
+	// If git repo is provided, create a separate js/ts file for script
+	var overwrite bool
+	if fields.GitCloneURL != "" {
+		scriptPath := path.Join(".brigade", brigadeScriptFileName)
+		if overwrite, err = checkFileOverwrite(scriptPath); err != nil {
+			return err
+		}
+		if overwrite {
+			if err = ioutil.WriteFile(
+				scriptPath,
+				scriptBytes,
+				0644,
+			); err != nil {
+				return err
+			}
+			fmt.Println("Creating " + brigadeScriptFileName + "...")
```

Comment:
Little bit of an issue here. This is going to say "creating" even if what we're doing is overwriting. We really should differentiate between those to things. This comment is applicable in several other spots as well.

Reply:
I had to throw in the print statements directly inside `checkFileOverwrite` due to the logic I used for checking for whether to write the file or not. Is that okay since `checkFileOverwrite` isn't being exported?

Reply:
I understand that your next comment addresses this problem a bit. Would it be better to return two booleans? One for indicating whether a file is being written at all, and whether it is being overwritten or created